tests/test_issue_maker.py

```diff
@@ -0,0 +1,161 @@
+        song = "Supersonics"
+        artist = "Caravan Palace"
+        generate_fake_unsupported()
+        del_line(song, artist)
+        with open('unsupported.txt', 'r') as f:
+            lines = f.readlines()
+            self.assertFalse(song + " by " + artist in lines)
+
+    @patch('requests.Response.json', return_value=sample_spotify_json)
+    @patch('requests.post', return_value=Response())
+    def test_update_token(self, requests_mock, json_mock):
+        from swaglyrics_backend.issue_maker import update_token
+        from swaglyrics_backend import issue_maker
+        update_token()
+        self.assertTrue(issue_maker.token != '')
+        self.assertTrue(issue_maker.t_expiry != 0)
+
+    @patch('requests.Response.json', return_value={'error': 'yes'})
+    @patch('requests.get', return_value=Response())
+    def test_check_song_returns_false_on_bad_response(self, requests_mock, response_mock):
+        from swaglyrics_backend.issue_maker import check_song
+        from swaglyrics_backend import issue_maker
+        issue_maker.t_expiry = time.time() + 3600
+        self.assertFalse(check_song("Miracle", "Caravan Palace"))
+
+    @patch('requests.Response.json')
+    @patch('requests.get', return_value=Response())
+    def test_that_check_song_returns_true(self, mock_get, mock_response):
+        from swaglyrics_backend.issue_maker import check_song
+        mock_response.return_value = get_correct_spotify_search_json('correct_spotify_data.json')
+        self.assertTrue(check_song("Miracle", "Caravan Palace"))
+
+    @patch('requests.Response.json', return_value=unknown_song_json)
+    @patch('requests.get', return_value=Response())
+    def test_that_check_song_returns_false_on_non_legit_song(self, mock_get, mock_response):
+        from swaglyrics_backend.issue_maker import check_song
+        self.assertFalse(check_song("Miracle", "Caravan Palace"))
+
+    @patch('requests.Response.json', return_value=None)
+    @patch('requests.get', return_value=Response())
+    def test_that_stripper_returns_none(self, mock_get, mock_response):
+        from swaglyrics_backend.issue_maker import genius_stripper
+        self.assertIsNone(genius_stripper("Miracle", "Caravan Palace"))
+
+    @patch('requests.Response.json')
+    @patch('requests.get')
+    def test_that_stripper_returns_stripper(self, mock_get, mock_response):
+        response = Response()
+        response.status_code = 200
+        mock_get.return_value = response
+        mock_response.return_value = get_correct_spotify_search_json('sample_genius_data.json')
+        from swaglyrics_backend.issue_maker import genius_stripper
+        self.assertEqual(genius_stripper("Miracle", "Caravan Palace"), "Caravan-palace-miracle")
+
+    def test_that_title_mismatches(self):
+        from swaglyrics_backend.issue_maker import is_title_mismatched
+        self.assertTrue(
+            is_title_mismatched({"Bohemian", "Rhapsody", "by", "Queen"}, "Miracle by Caravan Palace", 2))
```

Comment:
Why do you pass a set instead of a normal string?

Reply:
check `is_title_mismatched` function, it accepts words as first parameter and full title as second

Reply:
Right so actually we call split() which returns a list. The thing with set is that every element should be unique which may not exist for some songs. You should use a list here to mimic as close to the actual functioning

Reply:
https://github.com/SwagLyrics/swaglyrics-backend/blob/fda9ffe6345346d9da7c2dfb6dc895cc3cfdf7be/issue_maker.py#L148

Reply:
Didn't know that set must have unique values, I'll change that

Reply:
Yup.
![image](https://user-images.githubusercontent.com/27063113/73194173-c31f8280-4151-11ea-86db-9230b89a2c35.png)